### ISSUE TITLE
Website outline discussion/brainstoming

### ISSUE THREAD

**Issue by FFX01** [Cabal-Syndicate/randora_website#2]
Hi Guys.

So, to get right to it:
We need a game plan for the website development.

The first thing we need to do is come up with a list of basic features the website NEEDS to implement.

After that, we need to come up with a basic structure for the site. Basically, how are we going to write our views, db models, etc.

After that, we can start designing the U.I.

When that's all completed, we can start looking to implement additional features.

So basically, in this issue we should be discussing what the we want the website to be and to do, and how to implement that.

What are your thoughts?

**Comment by Master-Foo** [comment]
Thanks for being pro-active and bringing this up.

Here is my thinking. There are a couple of other guys working on this project, including you, and I, that makes four (perhaps more) that are working here. Not all of us are at the same level. Some are trying to just get a basic django tutorial under their belt. I've been doing Django for a years now. So, there is a wide gap in skill set.

Before we get too far into the specs of the site, I want everyone to have a basic foundational knowledge of Django and RESTful / MVC development, so everyone can contribute.

I created this repo to give people a place to start and ask questions. Eventually, we will get to the meat of the project.

What I'd like to see is, everyone complete the Django tutorial, so I can see where everyone is at and everyone can ask questions.

But you are correct, we can start planning features now. @FFX01 if you have previous experience with Django, I'd like you to take a proactive part in the planning of this site and guiding those who are new to REST / MVC. I'll also help, but it would be nice to have the "Official Django Guru".

Here's the game plan as far as I'm concerned.

The Website project exists for the purpose of supporting the Game Framework project.

So, immediate features would be hosting documentation, turorials, examples of peoples games, a message board, etc. basic things you'd find on any framework's website.

In later iterations I'd like to start phasing out GitHub as the place for colaboration and start adding functionality to our site which allows us to collaborate from there. This is in the far future though and I don't think we should be planning for any of that just yet.

All that being said. If someone completes the tutorial, that gives us an excellent place to start for all of these goals. We can use the tutorial as a spring board and work from there.

As far as starting plans for the UI. I don't think we should be spending a lot of time here. [Bootstap](http://getbootstrap.com/) is going to give us all the functionality we need. I don't think the cost / reward evaluation is favorable towards making a custom UI.

That being said, if someone REALLY wants to commit themselves to a custom UI, I support them. But let's get Bootstrap in place first, so we have something to work with and then add / replace with custom work later.

I'm open to other suggestions and disagreements.

**Comment by FFX01** [comment]
> All that being said. If someone completes the tutorial, that gives us an excellent place to start for all of these goals. We can use the tutorial as a spring board and work from there.

OK, so we should wait for someone to finish the tutorial and make a pull request to the repo. We can then merge it and begin work on the website functionality. It would be great if any of the people working on the tutorial could give us a status update on where they're at and a timeline of when they think they will finish.

The official Django tutorial has you create a polling app, so we should possibly integrate that into the final product because it's already built.

> I'll also help, but it would be nice to have the "Official Django Guru".

I definitely wouldn't call myself a guru by any means. In fact, I have more experience with Flask. However, I do have a good grasp on the fundamentals of MVC development.

> So, immediate features would be hosting documentation, turorials, examples of peoples games, a message board, etc. basic things you'd find on any framework's website.

OK, out of all of these, the most difficult and time consuming will be the implementation of the message boards because we will have to do a lot of work on the db side of things. The rest of the features should be fairly easy to implement with basic url routing and basic HTML templates. I feel like the first thing we should focus on besides the home/contact pages will be the documentation pages. Instead of having individual paths for each piece of documentation, I believe t would be a good idea to dynamically generate page content based on queries. What do you think?

> As far as starting plans for the UI. I don't think we should be spending a lot of time here. Bootstap is going to give us all the functionality we need. I don't think the cost / reward evaluation is favorable towards making a custom UI.

I agree. We could also look into Material Design Light as well.

**Comment by Master-Foo** [comment]
I'm going to give everyone else 24 hours to give an update on their progress. I don't expect anyone to finish in that time, but it would be nice to see some progress on their fork.

@FFX01 will you please create individual issues for planning things like the message board / documentation / etc? We can continue with details there.

I'm open to other UI like Material Design Light, however I only have experience with Bootstrap so I can't be of much help there. Open an issue for this as well so we can debate the pros and cons

**Comment by FFX01** [comment]
I'm on it.
